app/src/main/java/de/danoeh/antennapod/fragment/swipeactions/SwipeActions.java

```diff
@@ -0,0 +1,298 @@
+import de.danoeh.antennapod.fragment.QueueFragment;
+import de.danoeh.antennapod.model.feed.FeedItem;
+import de.danoeh.antennapod.model.feed.FeedItemFilter;
+import de.danoeh.antennapod.ui.common.ThemeUtils;
+import de.danoeh.antennapod.view.viewholder.EpisodeItemViewHolder;
+import it.xabaras.android.recyclerview.swipedecorator.RecyclerViewSwipeDecorator;
+
+public class SwipeActions {
+    public static final String PREF_NAME = "SwipeActionsPrefs";
+    public static final String PREF_SWIPEACTIONS = "PrefSwipeActions";
+    public static final String PREF_NO_ACTION = "PrefNoSwipeAction";
+
+    public static final List<SwipeAction> swipeActions = Collections.unmodifiableList(
+            Arrays.asList(new AddToQueueSwipeAction(), new RemoveFromInboxSwipeAction(),
+                    new StartDownloadSwipeAction(), new MarkFavouriteSwipeAction(),
+                    new MarkPlayedSwipeAction(), new RemoveFromQueueSwipeAction())
+    );
+
+    private RecyclerView recyclerView;
+    public ItemTouchHelper itemTouchHelper;
+    private final Fragment fragment;
+    private final String tag;
+    private FeedItemFilter filter = null;
+
+    private NewSwipeCallback newSwipeCallback = SimpleSwipeCallback::new;
+
+    public SwipeActions(Fragment fragment, String tag) {
+        this.fragment = fragment;
+        this.tag = tag;
+        setItemTouchHelper();
+    }
+
+    public void setFilter(FeedItemFilter filter) {
+        this.filter = filter;
+    }
+
+    public SwipeActions attachTo(RecyclerView recyclerView) {
+        this.recyclerView = recyclerView;
+        itemTouchHelper.attachToRecyclerView(recyclerView);
+        //reload as settings might have changed
+        fragment.getLifecycle().addObserver((LifecycleEventObserver) (source, event) -> {
+            if (event == Lifecycle.Event.ON_RESUME) {
+                resetItemTouchHelper();
+            }
+        });
+
+        return this;
+    }
+
+    public void setItemTouchHelper() {
+        itemTouchHelper = new ItemTouchHelper(newSwipeCallback.construct());
+    }
+
+    private static Pair<SwipeAction, SwipeAction> getPrefs(Context context, String tag, String defaultActions) {
+        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
+        String prefsString = prefs.getString(PREF_SWIPEACTIONS + tag, defaultActions);
+
+        String[] rightleft = prefsString.split(",");
+
+        //no preferences set, no default (very fist swipe)
+        if (rightleft.length != 2) {
+            return null;
+        }
+
+        Optional<SwipeAction> right = Stream.of(swipeActions)
+                .filter(a -> a.getId().equals(rightleft[0])).findFirst();
+        Optional<SwipeAction> left = Stream.of(swipeActions)
+                .filter(a -> a.getId().equals(rightleft[1])).findFirst();
+
+        // invalid ids
+        if (!right.isPresent() || !left.isPresent()) {
+            return new Pair<>(swipeActions.get(0), swipeActions.get(0));
```

Comment:
TODO. Before I added this line, the dialog crashed when the name of swipe actions changed. How about just storing each action in its own preference instead of storing both together and then having to split?

Reply:
Hmm I had a optional check before no? I think it's the same, splitting just saves a few lines for the preferences.